aea/decision_maker/base.py

```diff
@@ -301,11 +305,36 @@ def _handle_tx_message(self, tx_message: TransactionMessage) -> None:
         :param tx_message: the transaction message
         :return: None
         """
-        score_diff = self.preferences.get_score_diff_from_transaction(self.ownership_state, tx_message)
-        if score_diff >= 0.0:
-            pass  # TODO: implement tx flow
-            # envelope = Envelope()
-            # self.outbox.put(envelope)
+        amount = tx_message.get("amount")
+        m_address = self.generate_address_from_public_key(self._wallet.public_keys['fetchai'])
+        api = LedgerApi("127.0.0.1", 8100)
+        self._generate_wealth()
+        print(api.tokens.balance(m_address))
+        if amount <= api.tokens.balance(m_address):
+            m_entity = Entity.from_hex(self._wallet.crypto_objects['fetchai'].private_key)
+            to = self.generate_address_from_public_key(tx_message.get("counterparty"))
+            api.sync(api.tokens.transfer(m_entity, to, 100, 20))
+            print(api.tokens.balance(m_address))
+
+    def _generate_wealth(self) -> None:
+        """
+        Generate tokens to be able to make a transaction.
+
+        :return:
+        """
+        m_entity = Entity.from_hex(self._wallet.crypto_objects['fetchai'].private_key)
+        print(m_entity.public_key)
+        api = LedgerApi("127.0.0.1", 8100)
+        api.tokens.wealth(m_entity, 1000)
+
+    def generate_address_from_public_key(self, public_key) -> Address:
```

Comment:
This should not be needed here, see above.

Reply:
If you want such a helper method then I would move it into the crypto module.

Reply:
@DavidMinarsch  I could add this in the Wallet as a property. But is it something that we want? I mean there is the functionality in the fetch_crypto_base but we want something that specific in the wallet (It is only for the fetch public_key)?